testing/internal/e2e/tests/base/paginate_managed_group_test.go

```diff
@@ -0,0 +1,199 @@
+			"-format=json",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	var initialManagedGroups managedgroups.ManagedGroupListResult
+	err = json.Unmarshal(output.Stdout, &initialManagedGroups)
+	require.NoError(t, err)
+
+	var returnedIds []string
+	for _, managedgroup := range initialManagedGroups.Items {
+		returnedIds = append(returnedIds, managedgroup.Id)
+	}
+
+	require.Len(t, initialManagedGroups.Items, c.MaxPageSize+1)
+	assert.Empty(t, cmp.Diff(returnedIds, managedgroupIds, cmpopts.SortSlices(func(i, j string) bool { return i < j })))
+	assert.Empty(t, initialManagedGroups.ResponseType)
+	assert.Empty(t, initialManagedGroups.RemovedIds)
+	assert.Empty(t, initialManagedGroups.ListToken)
+
+	// Create a new managedgroup and destroy one of the other managed groups
+	newManagedGroupId := boundary.CreateNewManagedGroupApi(t, ctx, client, amId)
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"managed-groups", "delete",
+			"-id", initialManagedGroups.Items[0].Id,
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	// List again, should have the new managedgroup but not the deleted managed group
+	output = e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"managed-groups", "list",
+			"-auth-method-id", amId,
+			"-format=json",
+		),
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	var newManagedGroups managedgroups.ManagedGroupListResult
+	err = json.Unmarshal(output.Stdout, &newManagedGroups)
+	require.NoError(t, err)
+
+	require.Len(t, newManagedGroups.Items, c.MaxPageSize+1)
+	// The first item should be the most recently created, which
+	// should be our new managedgroup
+	firstItem := newManagedGroups.Items[0]
+	assert.Equal(t, newManagedGroupId, firstItem.Id)
+	assert.Empty(t, newManagedGroups.ResponseType)
+	assert.Empty(t, newManagedGroups.RemovedIds)
+	assert.Empty(t, newManagedGroups.ListToken)
+	// Ensure the deleted managedgroup isn't returned
+	for _, managedgroup := range newManagedGroups.Items {
+		assert.NotEqual(t, managedgroup.Id, initialManagedGroups.Items[0].Id)
+	}
+}
+
+// TestApiPaginateManagedGroups asserts that the API automatically paginates to retrieve
+// all managedgroups in a single invocation.
+func TestApiPaginateManagedGroups(t *testing.T) {
+	e2e.MaybeSkipTest(t)
+	c, err := loadTestConfig()
+	require.NoError(t, err)
+
+	client, err := boundary.NewApiClient()
+	require.NoError(t, err)
+	adminToken := client.Token()
+	ctx := context.Background()
+	sClient := scopes.NewClient(client)
+	amClient := authmethods.NewClient(client)
+	atClient := managedgroups.NewClient(client)
+	newOrgId := boundary.CreateNewOrgApi(t, ctx, client)
+	amId := boundary.CreateNewOidcAuthMethodApi(t, ctx, client, newOrgId)
+	t.Cleanup(func() {
+		ctx := context.Background()
+		client.SetToken(adminToken)
+		_, err := amClient.Delete(ctx, amId)
+		require.NoError(t, err)
+		_, err = sClient.Delete(ctx, newOrgId)
+		require.NoError(t, err)
+	})
```

Comment:
Should we have the org cleanup occur just after org creation just in case something happens to fail beforehand for whatever reason?